### ISSUE TITLE
No body for section-slide

### ISSUE THREAD

**Issue by Beiri22** [touying-typ/touying#79]
Hi, simple example:
```
= My Section 1
#my-custom-speaker-notes("something for section")

== My Slide
#my-custom-speaker-notes("something for slide")
```

It seems that the new-section-slide-fn does get the heading as body(first positional argument); but not the custom note that I provided. The heading can easily be obtained via utils.display-current-heading(level: 1); but the body is gone or triggering a new slide. As far as I understand, the part after "=My Section 1" could be treated as body, also for sections. 

What do you think?

**Comment by OrangeX4** [comment]
There is ambiguity here:

```typst
= Title

Slide body.
```

How many slides will be generated? touying currently believes that you should generate two slides instead of having the `Slide body` as part of the new-section-slide.

At the moment we don't have a very good syntax to distinguish between these.

**Comment by Beiri22** [comment]
But why would you create a second slide without a title? This is a personal preference, but I would not want that. 

As far as I understand:
```
= Section

Some Content
```

creates 2 slides: The section slide and the title-less slide with some content, whereas

```
== Title

Some Content
```

would create only one slide with title and content? This could be confusing - both use heading to make slides but handle content differently...

**Comment by OrangeX4** [comment]
I think your idea is reasonable. But I still think that the user who see

```typst
= Section

Some Content
```

will consider it should create 2 slides.

If only 1 slide is generated, then only a new-section-slide with invisible `Some Content` will be generated (at present, basically all themes believe that new-section-slide only has title, without considering body displaying). This is a huge annoyance for users, who will think that disappearing content is a bug.

**Comment by OrangeX4** [comment]
In other words, do you think it makes more sense to just register `slide-fn` and then let theme makers dispatch `new-section-slide-fn` themselves in `slide-fn`?

**Comment by Beiri22** [comment]
No, that’s not exactly what I meant. I would prefer to keep them as two separate functions, but both could take the body/content as a parameter. A section slide might also require a body—not just for metadata as I intend to use it, but also to add section-specific elements like an image or a quote.

**Comment by OrangeX4** [comment]
Okay, I think such a design is reasonable. There is just a small problem, if the `section-slide-uses-body` of built-in themes are configured with `false`, then your usage still won't work. Users may get a little confused.

```typst
= My Section 1
#my-custom-speaker-notes("something for section")
```

**Comment by OrangeX4** [comment]
In short, I agree with the idea that new-section-slide-fn can have a body, but for the syntax, I think it needs to be carefully designed. Thanks for your issue.

**Comment by Beiri22** [comment]
> Okay, I think such a design is reasonable. There is just a small problem, if the `section-slide-uses-body` of built-in themes are configured with `false`, then your usage still won't work. Users may get a little confused.

From what I understand, nearly all options can be overridden by the user through the config-xxx() methods. This means I could use the built-in template, override this specific setting within my presentation, and it should work as expected. Is that correct?

**Comment by OrangeX4** [comment]
I don't think it's that simple because `#let new-slide-section(body)` must accept a positional argument unless we use named argument `#let new-slide-section(body: none)`.

**Comment by Beiri22** [comment]
But ```#let new-slide-section()``` could accept an argument sink ```...args``` so if a body is provided it would silently go unter the positional part of the sink argument...

**Comment by Beiri22** [comment]
At least for me it would be great to use h1 and h2 for sections and slides while being able to use my annotations for both. Right now I would have to create all sections the explicit way... So I'm really interested in this. Do you consider allowing bodies for section-slides?

**Comment by OrangeX4** [comment]
I'll think about it, but it may take some time to think about the best way to do it.

**Comment by Beiri22** [comment]
Don't underestimate the power of discussion and talking. 😀 Is there any place to find interested users to incorporate into the discussion?

**Comment by OrangeX4** [comment]
You can discuss it directly under this issue, I just haven't had time to think about it recently. Maybe I'll add this feature in the next release.

**Comment by Beiri22** [comment]
One question. For now I would like to call #section-slide function explicitly, but doing so, no new  heading is generated und thus missing in the agenda. I think there is some trick to make it work, right?

**Comment by OrangeX4** [comment]
Minimal reproducible example? I can use the following code to generate it normally:

```typst
#show: university-theme.with(
  config-common(
    new-section-slide-fn: none,
  )
)

= Section

#new-section-slide[]
```

**Comment by OrangeX4** [comment]
In the next version, a `config-common(receive-body-for-new-section-slide-fn: true)` configuration will be added, which defaults to true.
